toolkits/microsoft/arcade_microsoft/outlook_calendar/tools/list_events_in_time_range.py

```diff
@@ -0,0 +1,56 @@
+from typing import Annotated
+
+from arcade.sdk import ToolContext, tool
+from arcade.sdk.auth import Microsoft
+from msgraph.generated.users.item.calendar.calendar_view.calendar_view_request_builder import (
+    CalendarViewRequestBuilder,
+)
+
+from arcade_microsoft.client import get_client
+from arcade_microsoft.outlook_calendar._utils import (
+    convert_timezone_to_offset,
+    create_timezone_request_config,
+    get_default_calendar_timezone,
+    replace_timezone_offset,
+)
+from arcade_microsoft.outlook_calendar.models import Event
+
+
+@tool(requires_auth=Microsoft(scopes=["MailboxSettings.Read", "Calendars.ReadBasic"]))
+async def list_events_in_time_range(
+    context: ToolContext,
+    start_date_time: Annotated[
+        str,
+        "The start date and time of the time range, represented in "
+        "ISO 8601 format. Timezone offset is ignored. For example, 2025-04-24T19:00:00",
+    ],
+    end_date_time: Annotated[
+        str,
+        "The end date and time of the time range, represented in "
+        "ISO 8601 format. Timezone offset is ignored. For example, 2025-04-24T19:30:00",
+    ],
+    limit: Annotated[int, "The maximum number of events to return. Max 1000. Defaults to 10"] = 10,
+) -> Annotated[dict, "A dictionary containing a list of events"]:
+    """List events in the user's calendar in a specific time range.
+
+    Ignores timezone offsets provided in the start_date_time and end_date_time parameters.
+    Instead, uses the user's default calendar timezone to filter events.
+    If the user has not set a timezone for their calendar, then the timezone will be UTC.
+    """
+    client = get_client(context.get_auth_token_or_empty())
+
+    time_zone = await get_default_calendar_timezone(client)
+    time_zone_offset = convert_timezone_to_offset(time_zone)
+    start_date_time = replace_timezone_offset(start_date_time, time_zone_offset)
```

Comment:
perhaps adding a datetime format validation before using the arg value? would allow to raise an error that is more descriptive of what needs to be corrected